docs/ref/models/fields.txt

```diff
@@ -552,8 +552,13 @@ be specified per app in :attr:`AppConfig.default_auto_field
 :ref:`automatic-primary-key-fields`.
 
 ``primary_key=True`` implies :attr:`null=False <Field.null>` and
-:attr:`unique=True <Field.unique>`. Only one primary key is allowed on an
-object.
+:attr:`unique=True <Field.unique>`.
+
+.. versionadded:: 5.2
+
+Only one field per model can set ``primary_key=True``. Composite primary keys
+must be defined using :class:`CompositePrimaryKey` instead of setting this flag
+to ``True`` for all fields to maintain this invariant.
```

Comment:
We should use `versionchanged` instead of `versionadded`, and `version*` annotations should be easy to remove, e.g.
```suggestion
:attr:`unique=True <Field.unique>`. Only one field per model can set
``primary_key=True``. Composite primary keys must be defined using
:class:`CompositePrimaryKey` instead of setting this flag to ``True`` for all
fields to maintain this invariant.

.. versionchanged:: 5.2

    The ``CompositePrimaryKey`` field was added.
```